### PR TITLE
Allow a specified address to disable/enable the Exchange 

### PR DESCRIPTION
**Pull request by m-chrzan**
### Description

For purposes of Baklava, we might need to periodically turn the Exchange off. This PR adds `freeze` and `unfreeze` functions to the Exchange, allowing an address specified at migration time (and modifiable via `setFreezer`) to disable the `exchange` function.

### Tested

Unit tests.

### Other changes

This PR adds a new helper contract, Freezable, that specifies a `freezer` address and provides modifiers to disable certain contract functions if the contract becomes "frozen".

This is in anticipation of a similar change needed in LockedGold.

### Related issues

- Begins to address #1431 

### Backwards compatibility

Not backwards compatible, changes Exchange storage layout.

### CODE REVIEW

File: packages/protocol/contracts/baklava/Freezable.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  // onlyFreezer functions can only be called by the specified `freezer` address
+  modifier onlyFreezer() {
+    require(msg.sender == freezer);
+    _;
+  }
+
+  // onlyWhenNotFrozen functions can only be called when `frozen` is false, otherwise they are
+  // noops.
+  modifier onlyWhenNotFrozen() {
```
Comment (asaj) [inline packages/protocol/contracts/baklava/Freezable.sol:16]: Is this necessary? I would consider cutting this function and renaming `onlyWhenNotFrozenOrThrow` to `onlyWhenNotFrozen`

**Review by asaj — APPROVED** [review]
Nice 👍

File: packages/protocol/contracts/baklava/Freezable.sol (hunk 1 above)
Comment (m-chrzan) [reply packages/protocol/contracts/baklava/Freezable.sol:16]: Should have cleared this up - this is already looking forward to the next change, where `distributeEpochPayments` will need to be frozen. By not reverting when that function is frozen, I think we'll be able to contain this temporary change for Baklava to smart contracts.

I could see an argument for reverting being the default semantic, happy to rename this one to `onlyWhenNotFrozenNoThrow`.

WDYT?

**Comment by asaj** [comment]
Did you mean to re-assign to me?

edit: Whoops, missed your comment

File: packages/protocol/contracts/baklava/Freezable.sol (hunk 1 above)
Comment (asaj) [reply packages/protocol/contracts/baklava/Freezable.sol:16]: Are you sure we don't want to revert when we freeze epoch rewards for TGCSO? Reverting would prevent any proof-of-stake action from happening in `Finalize`, e.g. validator score changes, payments, etc.

**Pull request merged**